yargs.js

```diff
@@ -397,6 +402,7 @@ function Yargs (processArgs, cwd, parentRequire) {
         opt = {}
       }
 
+      rawOptions[key] = opt
```

Comment:
one concern I have with this, is that there are other ways that parameters can be set on options, .e.g., using the chaining API:

```js
yargs
  .describe('a', 'my a option')
  .default('a', 99)
```
I think you would do better to use the method `yargs.getOptions`; which already exists, and is used for similar purposes in the utility libraries. This returns the options object which is ultimately executed by yargs' parser.

Do you think this would solve your use-case for you; if so, would you perhaps be willing to contribute some documentation, I would greatly appreciate it.